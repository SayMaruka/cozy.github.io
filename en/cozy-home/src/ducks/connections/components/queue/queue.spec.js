'use strict'

/* eslint-env jest */

import React from 'react'
import { configure, shallow } from 'enzyme'
import { tMock } from '../../../../../test/jestLib/I18n'

import { Queue } from './queue'

import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

describe('Queue component', () => {
  const queue = [
    {
      konnector: { slug: 'testk' },
      label: 'Test',
      status: 'done',
      triggerId: '0412d5795e464ead99f5cea2611bbf21'
    },
    {
      konnector: { slug: 'testk2' },
      label: 'Test 2',
      status: 'ongoing',
      triggerId: 'f042dd05b97842c3acfed33253009659'
    }
  ]

  it('should render', () => {
    const component = shallow(
      <Queue queue={queue} visible={false} t={tMock} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })
})
