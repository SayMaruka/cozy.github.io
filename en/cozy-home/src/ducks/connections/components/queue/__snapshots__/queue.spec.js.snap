// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queue component should render 1`] = `
<div
  className="queue"
>
  <h4
    className="queue-header"
    onDoubleClick={[Function]}
  >
    <div
      className="queue-header-inner"
    >
      <span
        className="coz-desktop"
      >
        Syncing accounts:
      </span>
      <span
        className="coz-mobile"
      >
        1 of 2
      </span>
    </div>
  </h4>
  <progress
    className="queue-progress"
    max={2}
    value={1}
  />
  <div
    className="queue-content"
  >
    <div
      className="queue-list"
    >
      <Item
        konnector={
          Object {
            "slug": "testk",
          }
        }
        label="Test"
        status="done"
        t={[Function]}
      />
      <Item
        konnector={
          Object {
            "slug": "testk2",
          }
        }
        label="Test 2"
        status="ongoing"
        t={[Function]}
      />
    </div>
  </div>
</div>
`;
